protos/helpers/client.go

```diff
@@ -0,0 +1,128 @@
+package helpers
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	defaultSchema = "http"
+)
+
+var (
+	domainRegexp = regexp.MustCompile(`^([a-zA-Z0-9-]{1,63}\.)*([a-zA-Z]{1,63})$`)
+	ipv4Regexp   = regexp.MustCompile(`^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$`)
+)
+
+type parsedURL struct {
+	schema string
+	host   string
+	port   int
+}
+
+func NewGRPCConn(ctx context.Context, address string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+	u, err := parse(address)
+	if err != nil {
+		return nil, err
+	}
+	if u.schema == "http" {
+		opts = append(opts, grpc.WithInsecure())
+	} else if u.schema == "https" {
+		h2creds := credentials.NewTLS(&tls.Config{NextProtos: []string{"h2"}})
+		opts = append(opts, grpc.WithTransportCredentials(h2creds))
+	} else {
+		return nil, errors.New("not supported schema")
+	}
+	return grpc.DialContext(ctx, fmt.Sprintf("%s:%d", u.host, u.port), opts...)
+}
+
+func parse(rawU string) (*parsedURL, error) {
```

Comment:
Why not use https://golang.org/pkg/net/url/#Parse? I think it's standard

Reply:
> Why not use https://golang.org/pkg/net/url/#Parse? I think it's standard

actually i used it, in line 54.
but it need more things to todo.
e.x. http://localhost:8080 will be parsed to `{schema: "http", host: "localhost:8080",port:"8080"}`

Reply:
ok